code/game/objects/items/weapons/stunbaton.dm

```diff
@@ -195,8 +195,18 @@
 		H.Jitter(10 SECONDS)
 		H.adjustStaminaLoss(stam_damage)
 
-	ADD_TRAIT(L, TRAIT_WAS_BATONNED, user_UID) // so one person cannot hit the same person with two separate batons
-	addtimer(CALLBACK(src, PROC_REF(baton_knockdown), L, user_UID, knockdown_duration), knockdown_delay)
+	/// check if the baton does in fact have a knockdown delay, in case of subtypes or admemery
+	if(knockdown_delay > 0)
+		ADD_TRAIT(L, TRAIT_WAS_BATONNED, user_UID) // so one person cannot hit the same person with two separate batons
+		L.apply_status_effect(STATUS_EFFECT_BATON_KNOCKDOWN)
+		var/datum/status_effect/baton_delayed_knockdown/stun_timer = L.has_status_effect(STATUS_EFFECT_BATON_KNOCKDOWN)
+		/// overrides the default status effect values in case of subtypes or var edits
+		if(istype(stun_timer))
+			stun_timer.knockdown_duration = knockdown_duration
+			stun_timer.duration = world.time + knockdown_delay
```

Comment:
you dont need to do istype. and would you make this handled in the on_creation like how normal incapacitating status effects do it?

Reply:
The knockdown duration and delay is determined by attributes of the baton itself (e.g. a stun prod only knocks down for 6 seconds), and the status effect's duration and delay need to be changed to match. There is no clean way I can think of for an on_creation to check the attributes of the baton that created it.

The istype is me worrying about edge cases where the status effect might not get created for whatever reason.

Reply:
If there's an edge case then it should runtime and then we can diagnose the issue and fix it

Reply:
For future reference I think it'd be nice if status effects could be passed arguments upon creation for things like this. But that's outside of the scope of this PR I feel.

Reply:
please remove the istype